packages/amplify-cli/src/__tests__/team-provider-migrate.test.ts

```diff
@@ -0,0 +1,124 @@
+    },
+    categories: {
+      auth: {
+        teamprovider1819bdce: {
+          hostedUIProviderCreds: '[{"ProviderName":"Facebook","client_id":"asdasdasdasd","client_secret":"asdasdasd"}]',
+        },
+      },
+    },
+  },
+  prod: {
+    awscloudformation: {
+      AuthRoleName: 'amplify-teamprovider-prod-164239-authRole',
+      UnauthRoleArn: 'arn:aws:iam::1234567891011:role/amplify-teamprovider-prod-164239-unauthRole',
+      AuthRoleArn: 'arn:aws:iam::1234567891011:role/amplify-teamprovider-prod-164239-authRole',
+      Region: 'us-east-1',
+      DeploymentBucketName: 'amplify-teamprovider-prod-164239-deployment',
+      UnauthRoleName: 'amplify-teamprovider-prod-164239-unauthRole',
+      StackName: 'amplify-teamprovider-prod-164239',
+      StackId: 'arn:aws:cloudformation:us-east-1:1234567891011:stack/amplify-teamprovider-prod-164239/1b625f60-18ae-11eb-9e65-0ab042f700a7',
+      AmplifyAppId: 'd1gmlw7l76gj9',
+    },
+    categories: {
+      auth: {
+        teamprovider1819bdce: {
+          hostedUIProviderCreds: '[{"ProviderName":"Facebook","client_id":"abc","client_secret":"adb"}]',
+        },
+      },
+    },
+  },
+};
+
+const teamProviderInfoWithoutSecrets = {
+  dev: {
+    awscloudformation: {
+      AuthRoleName: 'amplify-teamprovider-dev-134909-authRole',
+      UnauthRoleArn: 'arn:aws:iam::1234567891011:role/amplify-teamprovider-dev-134909-unauthRole',
+      AuthRoleArn: 'arn:aws:iam::1234567891011:role/amplify-teamprovider-dev-134909-authRole',
+      Region: 'us-east-1',
+      DeploymentBucketName: 'amplify-teamprovider-dev-134909-deployment',
+      UnauthRoleName: 'amplify-teamprovider-dev-134909-unauthRole',
+      StackName: 'amplify-teamprovider-dev-134909',
+      StackId: 'arn:aws:cloudformation:us-east-1:1234567891011:stack/amplify-teamprovider-dev-134909/df33f4d0-1895-11eb-a8b4-0e706f74ed45',
+      AmplifyAppId: 'd1gmlw7l76gj9',
+    },
+    categories: {
+      auth: {
+        teamprovider1819bdce: {},
+      },
+    },
+  },
+  prod: {
+    awscloudformation: {
+      AuthRoleName: 'amplify-teamprovider-prod-164239-authRole',
+      UnauthRoleArn: 'arn:aws:iam::1234567891011:role/amplify-teamprovider-prod-164239-unauthRole',
+      AuthRoleArn: 'arn:aws:iam::1234567891011:role/amplify-teamprovider-prod-164239-authRole',
+      Region: 'us-east-1',
+      DeploymentBucketName: 'amplify-teamprovider-prod-164239-deployment',
+      UnauthRoleName: 'amplify-teamprovider-prod-164239-unauthRole',
+      StackName: 'amplify-teamprovider-prod-164239',
+      StackId: 'arn:aws:cloudformation:us-east-1:1234567891011:stack/amplify-teamprovider-prod-164239/1b625f60-18ae-11eb-9e65-0ab042f700a7',
+      AmplifyAppId: 'd1gmlw7l76gj9',
+    },
+    categories: {
+      auth: {
+        teamprovider1819bdce: {
+          hostedUIProviderCreds: '[{"ProviderName":"Facebook","client_id":"abc","client_secret":"adb"}]',
+        },
+      },
+    },
+  },
+};
+const secrets = {
+  d1gmlw7l76gj9: {
```

Comment:
These are 

> > > Overall the approach looks ok, but I have a concern about connecting the secrets stuff to the AmplifyAppId which in case of 25+ Apps will not be present and I think the current implementation would fail. since we don't have a project id or something like that, what if we would use the root stack's resource Id (last part of the ARN after the /), that is guaranteed to be unique by CF, right?
> > > Based on this the suggested new JSON structure would have the `rootStackId` instead of `amplifyAppId`, and we could lose the environment name at all if we want since rootStackId would give a different granularity to the secrets.
> > > A requested also changes around `StateManager` and tests.
> > > Proposed structure with root stack id:
> > > ```json
> > > {
> > >     "appSecrets": [
> > >         {
> > >             "rootStackId": "<uuid>",
> > >             // However it will be always a single env this way, but keep for portable future changes
> > >             "environments": {
> > >                 "dev": {
> > >                     "auth": {
> > >                         "teamprovider1819bdce": {
> > >                             "hostedUIProviderCreds": "[{'ProviderName':'Facebook','client_id':'asdasdasdasd','client_secret':'asdasdasd'}]",
> > >                         }
> > >                     }
> > >                 }
> > >             }
> > >         }
> > >     ]
> > > }
> > > ```
> > 
> > 
> > I think the 25 AppId limit is an acceptable limit risk. It would be difficult to change the design after it has been approved and implemented
> 
> Even if we stick with using appId as the identifier, I would still like the json to be structured as Attila mentioned. To me having some random string of characters be a key in the json just looks messy and it doesn't convey any meaning

I will try but this will be difficult seeing that deadlines is really close and this requires a change through most of the implementation. This would have been the two design meeting we had regarding this.